demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -471,6 +490,33 @@ static void prvAzureDemoTask( void * pvParameters )
 }
 /*-----------------------------------------------------------*/
 
+#if CONFIG_ESP_TLS_USE_SECURE_ELEMENT
+/**
+ * @brief Get the serial number of the ATECC608 chip and the registration ID that will be used
+ * by the DPS client
+ * 
+ */
+    static uint32_t prvPrepareRegistrationIdFromATECC608(uint8_t *sernum, char *registration_id_string) {
```

Comment:
It's unclear if this is the correct registration id in all cases: if a custom certificate is used, can it have a different Subject CN name?

Reply:
Yes, that is correct. It can have a different subject CN name. This implementation would give the correct registration ID when using Trust&GO devices or TrustFLEX devices only. Most people would opt for either one of these owing to the simplicity of using these variants.

Do you think developers should be allowed two methods for registrationID preparation based on whether the device is a Trust&GO-TrustFLEX or the custom variant? If it is the former, use this function, if not, define your own and plug it in?

Reply:
> TrustFLEX devices only

See above comment: my understanding is that TrustFLEX can change as well - is this correct?

> Do you think developers should be allowed two methods for registrationID preparation based on whether the device is a Trust&GO-TrustFLEX or the custom variant?

I think it's an interesting option but due to complications with TrustFLEX and TrustCUSTOM it may be safer to fall-back to the original Azure IoT config (through ESP-IDF). We could have documentation on how to call this function to prepare the registrationID from ATECC.

Reply:
> See above comment: my understanding is that TrustFLEX can change as well - is this correct?

Yes, this is correct - it <ins>*can*</ins> change.

> We could have documentation on how to call this function to prepare the registrationID from ATECC.

This sounds like a good compromise. So in a nutshell, is this what you are leaning towards?
- If someone uses a TNGTLS chip, a function kicks in which over-writes the Azure IoT Config with **sn01223....01**.
- If someone uses a TFLXTLS or TCSM chip, they have to decide whether they want to retain our TNGTLS function or write their own. 
Either way - a new set of questions will pop up though... :-)